drop-down-terminal-x@bigbn.pro/terminal.js

```diff
@@ -525,18 +529,39 @@ const DropDownTerminalX = new Lang.Class({
 
     tab.container.show()
     tab.terminal.show()
+    
+    // get the current working directory before changing the page
+    const currenWorkingDir = this._getCurrentWorkingDirectory()
+
     this.notebook.set_current_page(this.notebook.get_n_pages() - 1)
 
     this._updateFont(tab)
     this._updateBehaviour(tab)
     this._updateCustomCommand(tab)
     this._addUriMatchers(tab)
 
-    this._forkUserShell(tab.terminal, commandArgs)
+    this._forkUserShell(tab.terminal, currenWorkingDir, commandArgs)
     this._updateFocusMode(tab)
     return tab
   },
 
+  _getCurrentWorkingDirectory() {
+
+    let workingDir = GLib.get_home_dir()
+    
+
+    if(this.tabs.length === 0 || !this._openNewTerminalInCurrentDirectory) {
+      return workingDir
+    }
+
+    const currentTerminal = this.tabs[this.notebook.get_current_page()].terminal
+    
+    workingDir = currentTerminal.get_current_directory_uri() || GLib.get_home_dir()
+    
+    // get_current_directory_uri returns an uri like file://hostname/home/username. So parse it to a path.
+    return workingDir.replace(new RegExp(/file:\/\/.*?\/(.*)/g), '/$1')
```

Comment:
I'm trying to avoid using regExps everywhere where it is possible. Because they are slow and usualy complicated.
There is much faster and simple solution available: https://jsperf.com/rexpvssplit/1